packages/lib/HtmlToMd.ts

```diff
@@ -26,8 +27,21 @@ export default class HtmlToMd {
 		turndown.use(turndownPluginGfm);
 		turndown.remove('script');
 		turndown.remove('style');
+		const pdfRule = {
+			filter: ['embed', 'object'],
+			replacement: function(_content: string, node: any, _options: any) {
+				if (node.getAttribute('src') && (node.getAttribute('type') === 'application/pdf' || node.getAttribute('src').endsWith('.pdf'))) {
+					// We are adding _pdf: prefix so that we can later distingish them from normal links and create resources for them.
+					return `[${node.getAttribute('src')}](_pdf:${node.getAttribute('src')})`;
+				}
+				return '';
+			},
+		};
+		if (options.loadPdfs) {
+			turndown.addRule('pdf', pdfRule);
+		}
 		let md = turndown.turndown(html);
-		if (options.baseUrl) md = markdownUtils.prependBaseUrl(md, options.baseUrl);
+		if (options.baseUrl) md = markdownUtils.prependBaseUrl(md, options.baseUrl, (url: string) => options.loadPdfs ? !url.startsWith('_pdf:') : true);
```

Comment:
Any reason why you need this? This code is there so that relative URLs are converted to absolute ones, which makes it possible to download the assets. If you skip that for PDFs, your code will fail to download the PDFs files with relative URLs.

Reply:
The clipper has code to convert relative urls to absolute ones before sending so this ended up working but I agree that we should have a better approach here, maybe: remove the marker, let the func do its job then prepend it again. Will look at it.